dgraphHelper.js

```diff
@@ -0,0 +1,370 @@
+import diff from 'deep-diff';
+import dgraph from 'dgraph-js';
+import grpc from 'grpc';
+
+import { SCHEMA, TYPES } from './schema';
+
+const DB_URL = process.env.DGRAPH_URL;
+
+const create_client = () => {
+  const CLIENT_STUB = new dgraph.DgraphClientStub(DB_URL, grpc.credentials.createInsecure());
+  return new dgraph.DgraphClient(CLIENT_STUB);
+}
+
+const get_schema = async client => (await client.newTxn().query('schema {}')).getJson();
+
+const prepare_value_string = predicate => {
+  const PREPARED_ARRAY = predicate.split(' ');
+  if (PREPARED_ARRAY[PREPARED_ARRAY.length - 1] === '.') {
+    PREPARED_ARRAY.pop();
+  } else {
+    throw new Error('Missing . at the end of predicate, or incorrect spacing.');
+  }
+  return PREPARED_ARRAY;
+};
+
+const type_check = (type, object) => {
+  const NORMAL_TYPES = ['default', 'bool', 'datetime', 'float', 'geo', 'int', 'password', 'string', 'uid'];
+  const LIST_TYPES = ['[default]', '[bool]', '[datetime]', '[float]', '[geo]', '[int]', '[string]', '[uid]'];
+  const TYPE_IS_NOT_ALIST = NORMAL_TYPES.some(value => value === type);
+  const TYPE_IS_LIST = LIST_TYPES.some(value => value === type);
+  if (!TYPE_IS_NOT_ALIST && !TYPE_IS_LIST) {
+    throw new Error('Incorrect or missing type in predicate.');
+  } else if (TYPE_IS_LIST) {
+    object.type = type.slice(1, -1);
+  } else {
+    object.type = type;
+  }
+  if (TYPE_IS_LIST) {
+    object.list = true;
+  }
+};
+
+const index_check = (aValues, object) => {
+  const TOKENIZER_ARRAY = [];
+  const INDEX = aValues.some(value => {
+    if (value.includes('@index')) {
+      if (value.slice(6, 7) !== '(' || value.slice(-1) !== ')') {
+        throw new Error('@index is invalid, missing parenthesis or there are spaces in tokenizer.');
+      }
+      const FIELDS = value.slice(7, -1).split(',');
+      FIELDS.forEach(field => {
+        TOKENIZER_ARRAY.push(field.trim());
+      });
+      return true;
+    }
+    return false;
+  });
+  if (INDEX) {
+    object.index = INDEX;
+    object.tokenizer = TOKENIZER_ARRAY;
+  }
+};
+
+const other_options = (aValues, object) => {
+  aValues.forEach(value => {
+    if (value.includes('@upsert')) {
+      object.upsert = true;
+    } else if (value.includes('@lang')) {
+      object.lang = true;
+    }
+  })
+}
+
+const create_json_schema = () => {
+  const JSON_SCHEMA = [];
+  Object.entries(SCHEMA).forEach(([key, value]) => {
+    const OBJECT = { predicate: key };
+    const PREDICATES_ARRAY = prepare_value_string(value);
+    try {
+      type_check(PREDICATES_ARRAY[0], OBJECT);
```

Comment:
type_check seems to be only used here, I don't think you need to separate it in another function